tests/end_to_end/andor_mixed/irix-o2-out.c

```diff
@@ -1,23 +1,36 @@
-s32 test(s32 arg0, s32 arg1, s32 arg2, s32 arg3); // static
+? test(s32 arg0, s32 arg1, s32 arg2, s32 arg3); // static
 
-s32 test(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
-    s32 phi_v1;
+? test(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+    s32 phi_a0;
+    ? phi_v1;
 
-    if ((arg0 + arg1) == 0) {
-        if (((arg1 + arg2) == 0) || ((arg0 * arg1) == 0)) {
-            phi_v1 = 0;
-            if (arg3 != 0) {
-                phi_v1 = 0;
-                if (arg0 != 0) {
-block_5:
-                    phi_v1 = arg0 + arg3;
-                }
-            }
+    if (((arg0 + arg1) != 0) || (((arg1 + arg2) != 0) && ((arg0 * arg1) != 0)) || ((phi_v1 = 0, (arg3 != 0)) && (phi_v1 = 0, (arg0 != 0)))) {
+        phi_v1 = 1;
+    }
+    if ((arg0 != 0) && ((arg1 != 0) || (arg2 != 0)) && ((arg3 != 0) || ((arg0 + 1) != 0))) {
+        phi_v1 = 2;
+    }
+    if (((arg0 != 0) && (arg3 != 0)) || (((arg1 != 0) || (arg2 != 0)) && ((arg0 + 1) != 0))) {
+        phi_v1 = 3;
+    }
+    if ((arg0 != 0) && (arg1 != 0) && ((arg2 != 0) || (arg3 != 0)) && (((arg0 + 1) != 0) || ((arg1 + 1) != 0))) {
+        phi_v1 = 4;
+    }
+    if ((((arg0 != 0) || (arg1 != 0)) && (arg2 != 0)) || ((arg3 != 0) && ((arg0 + 1) != 0)) || ((arg1 + 1) != 0) || ((arg2 + 1) != 0)) {
+        phi_v1 = 5;
+    }
+    if (((arg0 == 0) || (arg1 == 0)) && ((arg2 != 0) && (arg3 != 0) && (((arg0 + 1) != 0) || ((arg1 + 1) != 0)))) {
+        phi_v1 = 6;
+    }
```

Comment:
Crap - this one is not equivalent!
Original:
`if (((a && b) || (c && d)) && ((a + 1) || (b + 1))) `

Reply:
Hah, good catch! I wonder if we should do some fuzz-testing here.